File: 2021/01/01.py
def input_read(filename: str):
    with open(filename, "r") as f:
        input_list = [line.strip() for line in f]
    return input_list


input_list = input_read("input.txt")

int_input_list = [int(x) for x in input_list]

count = 0
prev = int_input_list[0]
for i in int_input_list[1:]:
    if i > prev:
        count += 1
    prev = i
print(count)

count = 0
previous_window_sum = 0
for i in range(len(int_input_list) - len(int_input_list) % 3):
    window = int_input_list[i : i + 3]
    # print(window)
    window_sum = sum(window)
    if window_sum > previous_window_sum:
        count += 1
    previous_window_sum = window_sum
print(count - 1)

File: 2021/01/01.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func main() {
	input := readInput("input.txt")
	var inputInt []int

	// convert contents to int
	for _, s := range input {
		i, err := strconv.Atoi(s)
		if err != nil {
			fmt.Fprint(os.Stderr, err)
			os.Exit(1)
		}
		inputInt = append(inputInt, i)
		// fmt.Printf("%v\n", i)
	}

	var prev int
	var sum int
	for _, x := range inputInt {
		if x > prev {
			sum++
		}
		prev = x
	}
	fmt.Printf("%v\n", sum-1)

	prev = 0
	sum = 0
	for i := range inputInt[:len(inputInt)-len(inputInt)%3] {
		windowSum := inputInt[i] + inputInt[i+1] + inputInt[i+2]
		if windowSum > prev {
			sum++
		}
		prev = windowSum
	}

	fmt.Printf("%v\n", sum-1)

}

func readInput(filepath string) []string {
	f, err := os.Open(filepath)
	if err != nil {
		fmt.Fprint(os.Stderr, err)
	}
	defer f.Close()

	scanner := bufio.NewScanner(f)
	var contents []string
	for scanner.Scan() {
		contents = append(contents, scanner.Text())
	}
	return contents
}
